File: CardMasterSkin/GraphicsElements/GraphicElement.vb
Imports System.Drawing
Imports CardMasterSkin.Skins

Namespace GraphicsElement

    Public MustInherit Class GraphicElement

        Public MustOverride Sub DrawShadow(g As Graphics, shadow As SkinShadow)
        Public MustOverride Sub Draw(g As Graphics)

    End Class

End Namespace

File: CardMasterSkin/GraphicsElements/GraphicElement.cs
using CardMasterSkin.Skins;
using System.Drawing;

namespace CardMasterSkin.GraphicsElements
{
    public abstract class GraphicElement
    {
        public abstract void DrawShadow(Graphics g, SkinElementShadow shadow);
        public abstract void Draw(Graphics g);
        public abstract void DrawBorder(Graphics g, SkinElementBorder border);

    }
}
